examples/plot_mandrill.py

```diff
@@ -0,0 +1,91 @@
+"""
+DiCoDiLe on the Mandrill image
+==============================
+
+This example illlustrates reconstruction of `Mandrill image
+<http://sipi.usc.edu/database/download.php?vol=misc&img=4.2.03>`_
+using DiCoDiLe algorithm with soft_lock "corner" and 9 workers.
+
+"""  # noqa
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from dicodile.data.images import fetch_mandrill
+
+from dicodile.utils.dictionary import init_dictionary
+from dicodile.utils.viz import display_dictionaries
+from dicodile.utils.csc import reconstruct
+
+from dicodile import dicodile
+
+
+###############################################################################
+# We will first download the Mandrill image.
+
+X = fetch_mandrill()
+
+plt.axis('off')
+plt.imshow(X.swapaxes(0, 2))
+
+
+###############################################################################
+# We will create a random dictionary of **K = 25** patches of size **8x8**
+# from the original Mandrill image to be used for sparse coding.
+
+# set dictionary size
+n_atoms = 25
+
+# set individual atom (patch) size
+atom_support = (4, 4)
+
+D_init = init_dictionary(X, n_atoms, atom_support, random_state=60)
+
+###############################################################################
+# We are going to run `dicodile` with **9** workers on **3x3** grids.
+
+# number of splits along each dimension
+w_world = 3
+
+# number of workers
+n_workers = w_world * w_world
+
+###############################################################################
+# Run `dicodile`.
+
+pobj, times, D_hat, z_hat = dicodile(X, D_init, n_iter=3,
+                                     n_workers=n_workers,
+                                     strategy='greedy',
+                                     dicod_kwargs={"max_iter": 10000},
+                                     verbose=6)
```

Comment:
- Here it would be good to explain n_iter and max_iter?
- Changing dicodile signature not to use dicod_kwargs might be more user friendly as we do not expose dicod. What do you think?
- How can we specify soft_lock? We say we are using soft_lock=corner at the top, but dicodile uses "border" inside.

Reply:
* Yes explaining the difference would be nice.
* I think you can remove `dicod_kwargs` here as the example converge in less iterations I think. This will simplify the example. Then I think in an another PR, we should make sure to set the right default and expose only interesting arguments but in another PR.
* Same as before. I would say let's just use the default (and mention this default) and see in another PR how to expose this.

Reply:
- If I do not misunderstand: 
   n_iter: the number of iterations for dicodile
   max_iter: the number of iterations for csc (dicodile_z). 
   
   Is this explanation ok, or would you prefer to explain differently?
   
- I would suggest to keep max_iter in dicod_kwargs. With max_iter=10000 it runs in 43s whereas with default value it runs in 999s.
- Ok. I just mention default soft_lock.

- Shall we also put 
  strategy: coordinate selection strategy for coordinate descent

Reply:
* the explanations seems ok for the 3 args.
*  Ok let's keep it that way.